### PR TITLE
Kafka 4060 remove zk client dependency in kafka streams followup

### PR DESCRIPTION
**Pull request by hjafarpour**
Updated the KAFKA-4060 code based on the new Admin client API. 
Also we won't delete internal topics anymore.

### CODE REVIEW

**Comment by hjafarpour** [comment]
@guozhangwang . Would you please review this PR.

**Comment by guozhangwang** [comment]
@hjafarpour Could you rebase?

**Comment by mjsax** [comment]
@hjafarpour Can you please avoid all those whitespace changes...

**Comment by guozhangwang** [comment]
retest this please

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java [hunk 1]
```diff
@@ -99,24 +101,22 @@ public void close() {
...
      * @param topicsMetadata
-     * @return
      */
-    private Map<InternalTopicConfig, Integer> getTopicsToBeDeleted(final Map<InternalTopicConfig, Integer> topicsPartitionsMap, Collection<MetadataResponse.TopicMetadata> topicsMetadata) {
+    private void validateTopicPartitons(final Map<InternalTopicConfig, Integer> topicsPartitionsMap, Collection<MetadataResponse.TopicMetadata> topicsMetadata) {
         Map<String, Integer> existingTopicNamesPartitions = getExistingTopicNamesPartitions(topicsMetadata);
-        Map<InternalTopicConfig, Integer> deleteTopics = new HashMap<>();
-        // Add the topics that don't exist to the nonExistingTopics.
         for (InternalTopicConfig topic: topicsPartitionsMap.keySet()) {
             if (existingTopicNamesPartitions.get(topic.name()) != null) {
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java:114]: nit: could we merge this two ifs to `if(XX.contains(name) && XX.get(name) != ...)`?

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java [hunk 2]
```diff
@@ -91,22 +91,22 @@ public void setUp() throws Exception {
     private TopicPartition t3p2 = new TopicPartition("topic3", 2);
 
     private List<PartitionInfo> infos = Arrays.asList(
-        new PartitionInfo("topic1", 0, Node.noNode(), new Node[0], new Node[0]),
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java:94]: All the changes on this file seems to be coding style only, are they intentional?

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignorTest.java [hunk 3]
```diff
@@ -131,7 +131,7 @@ public void testSubscription() throws Exception {
         String clientId = "client-id";
         UUID processId = UUID.randomUUID();
         StreamThread thread = new StreamThread(builder, config, new MockClientSupplier(), "test", clientId, processId, new Metrics(), Time.SYSTEM, new StreamsMetadataState(builder, StreamsMetadataState.UNKNOWN_HOST),
-                                               0) {
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignorTest.java:134]: All the changes on this file seems to be coding style only, are they intentional?

**Review by guozhangwang — COMMENTED** [review]
@hjafarpour Could you add unit test covering this behavior change?

**Comment by guozhangwang** [comment]
@hjafarpour Could you close this PR?

**Pull request closed without merging**